herbert_core/utilities/general/replace_inf_range.m

```diff
@@ -0,0 +1,46 @@
+function img_range = replace_inf_range(img_range)
+%REPLACE_INF_RANGE Remove old inf integration ranges which may be present
+% in old sqw or dnd objects
+%
+% Inputs:
+% img_range  -- 2xN array of min/max values with possible +-inf values
+%               present in it
+% Returns:
+% img_range  -- the same array of ranges, modifies so that inf ranges are
+%               replaced by closest defined value e.g:
+%  if input is:
+%  [ -1,-inf,-inf, 10,...
+%   inf, 1,   inf, 20]
+% The output would be
+%  [-1    ,1-eps,-eps, 10,...
+%   -1+eps,1    , eps, 20]
+% Where eps is smallest positive compatible with 1 i.e. 1+eps ~= 1
+%
+% Horace-4 does not support inf ranges any more.
+% Unlike previous Horace version, where  -inf:inf mean whole integration
+% range, dnd with 0-integration range is meaningless for dnd obj,
+% so will set +-0
```

Comment:
Not clear about this. The old way had -inf:inf as the whole integration range. What is the representation of the whole integration range in this new method? Using -eps for -inf in the above example suggests that negative ranges are not possible. Is this the case?

Reply:
hopefully clarified